### PR TITLE
Add support for defaulting Prow monitoring config, decouple from prow.k8s.io, and update README port-forwarding instructions to include grafana.

### PR DESCRIPTION
**Pull request by cjwagner**
/assign @chaodaiG 
/cc @stevekuznetsov @alvaroaleman 

From now on we should avoid hardcoding values specific to the prow.k8s.io instance and instead add new config fields as needed.  Config defaulting logic and helper functions can be added to the `defaultConfig` function in the `config_util.libsonnet` file.
New config fields should provide any defaulting logic needed to ensure that prometheus rules and grafana dashboards that use them behave well when no value is specified. This will allow us to safely update the monitoring stack for different instances by simply copying the latest rules and dashboards from upstream without having to update config.

### CODE REVIEW

**Review by papuneftali — APPROVED** [review]

**Review by papuneftali — APPROVED** [review]

File: config/prow/cluster/monitoring/mixins/lib/config.libsonnet [hunk 1]
```diff
@@ -1,80 +1,77 @@
...
+  // How long we go during work hours without seeing a webhook before alerting.
+  webhookMissingAlertInterval: '10m',
+
+  // How many days prow hasn't been bumped.
+  prowImageStaleByDays: 7,
+};
+
+// Generate the real config by adding in constant fields and defaulting where needed.
+{
+  _config+:: util.defaultConfig(util.consts + config),
```
Comment (chaodaiG) [inline config/prow/cluster/monitoring/mixins/lib/config.libsonnet:75]: Nit, this seems a bit excessive to me to call method from another module with the value from the other module, is there a way to make `defaultConfig` always use `consts`?

**Comment by chaodaiG** [comment]
Overall looks good to me, left a nit

/lgtm
/approve
/hold
If you want to deal with my nit

File: config/prow/cluster/monitoring/mixins/lib/config.libsonnet (hunk 1 above)
Comment (cjwagner) [reply config/prow/cluster/monitoring/mixins/lib/config.libsonnet]: Yeah thats part of the reason I moved the constants to the new file. I forgot to change this. Fixed.

**Pull request merged**